packages/api-clients/open-api/m2mGatewayApi.yml

```diff
@@ -0,0 +1,4833 @@
+      summary: List agreements
+      operationId: getAgreements
+      parameters:
+        - in: query
+          name: states
+          required: false
+          schema:
+            type: array
+            items:
+              $ref: "#/components/schemas/AgreementState"
+            default: []
+          explode: false
+        - in: query
+          name: producerIds
+          required: false
+          schema:
+            type: array
+            items:
+              type: string
+              format: uuid
+            default: []
+          explode: false
+        - in: query
+          name: consumerIds
+          required: false
+          schema:
+            type: array
+            items:
+              type: string
+              format: uuid
+            default: []
+          explode: false
+        - in: query
+          name: eserviceIds
+          schema:
+            type: array
+            items:
+              type: string
+              format: uuid
+            default: []
+          explode: false
+        - in: query
+          name: offset
+          required: true
+          schema:
+            type: integer
+            format: int32
+            minimum: 0
+        - in: query
+          name: limit
+          required: true
+          schema:
+            type: integer
+            format: int32
+            minimum: 1
+            maximum: 50
+      responses:
+        "200":
+          description: Agreements retrieved
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Agreements"
+          headers:
+            X-Rate-Limit-Limit:
+              $ref: '#/components/headers/RateLimitLimitHeader'
+            X-Rate-Limit-Remaining:
+              $ref: '#/components/headers/RateLimitRemainingHeader'
+            X-Rate-Limit-Interval:
+              $ref: '#/components/headers/RateLimitIntervalHeader'
+        "400":
+          description: Bad Request
+          content:
+            application/problem+json:
+              schema:
+                $ref: "#/components/schemas/Problem"
+          headers:
+            X-Rate-Limit-Limit:
+              $ref: '#/components/headers/RateLimitLimitHeader'
+            X-Rate-Limit-Remaining:
+              $ref: '#/components/headers/RateLimitRemainingHeader'
+            X-Rate-Limit-Interval:
+              $ref: '#/components/headers/RateLimitIntervalHeader'
+        "401":
+          description: Unauthorized
+          content:
+            application/problem+json:
+              schema:
+                $ref: "#/components/schemas/Problem"
+          headers:
+            X-Rate-Limit-Limit:
+              $ref: '#/components/headers/RateLimitLimitHeader'
+            X-Rate-Limit-Remaining:
+              $ref: '#/components/headers/RateLimitRemainingHeader'
+            X-Rate-Limit-Interval:
+              $ref: '#/components/headers/RateLimitIntervalHeader'
```

Comment:
Could you double-check that every route returns `400`, `401`, `403`, and `429`? There are codes that we should list in all routes because we check params, authentication, authorization, and rate limiting everywhere.
For example, here I see that 403 is missing

Reply:
GET routes are open to "m2m" and "m2m-admin", a 403 shouldn't really occur for authenticated users with these roles. Maybe we should remove the GET's 403s, WDYT?

Reply:
It could occur! If you call with a UI token for example...
It's true that it's checked by the audience during token verification, theoretically, but if the audience is misconfigured and you call the route with `"api"` or another user role, then it will return 403.

No strong opinion, since the audience check should make sure we call only with m2m/m2m-admin before the check on roles, we can also remove 403 from all GET calls and just leave 401

Reply:
IMHO, if the audience is misconfigured then the token should not be valid and throw a 401

Reply:
It's what already happens if it's misconfigured in the token :) But, if it's misconfigured in the service config, and we accept for example also `/internal`, then Internal tokens would pass and we would then get a 403 on role validation... it's an edge case so we could reasonably assume it's correctly configured, and keep only `401` in GET routes as default 👌

Let's just remember to add 403 where needed later - for example, GET /agreements/:agreementId returns 403 in case the requester is not the producer/consumer/delegate... we can add these more specific codes ad-hoc while implementing each route later